src/main/java/org/radarbase/management/web/rest/SourceResource.java

```diff
@@ -77,7 +85,9 @@
     public ResponseEntity<SourceDTO> createSource(@Valid @RequestBody SourceDTO sourceDto)
             throws URISyntaxException, NotAuthorizedException {
         log.debug("REST request to save Source : {}", sourceDto);
-        checkPermission(getJWT(servletRequest), SOURCE_CREATE);
+        MinimalProjectDetailsDTO project = sourceDto.getProject();
+        String projectName = project != null ? project.getProjectName() : null;
+        checkPermissionOnProject(getJWT(servletRequest), SOURCE_CREATE, projectName);
```

Comment:
This means a source cannot be created without being assigned to a project? Functionally sounds good to me. We may have to make the project required on the UI under entities/sources/create new source (until we remove that feature).

Reply:
If the project is not valid (like `null`), `checkPermissionOnProject` will only allow sys admins and oauth clients. That is fine for this use case.